adobe_vipm/adobe/errors.py

```diff
@@ -4,85 +4,121 @@
 from functools import wraps
 from typing import ParamSpec, TypeVar
 
-from requests import HTTPError, JSONDecodeError
+from requests import HTTPError
 
-Param = ParamSpec("Param")
+Param = ParamSpec("Param")  # noqa: WPS110
 RetType = TypeVar("RetType")
 
 logger = logging.getLogger(__name__)
 
 
 class AdobeError(Exception):
-    pass
+    """Basic Adobe Client Error."""
 
 
 class AdobeProductNotFoundError(AdobeError):
-    pass
+    """Product not found in the configuration."""
 
 
 class AuthorizationNotFoundError(AdobeError):
-    pass
+    """Authorization not found in configuration."""
 
 
 class ResellerNotFoundError(AdobeError):
-    pass
+    """Reseller not found."""
 
 
 class CountryNotFoundError(AdobeError):
-    pass
+    """Country is not found."""
 
 
 class CustomerDiscountsNotFoundError(AdobeError):
-    pass
+    """Customer benefits are not found for the customer."""
 
 
 class SubscriptionNotFoundError(AdobeError):
-    pass
+    """Subscription not found."""
 
 
 class SubscriptionUpdateError(AdobeError):
-    pass
+    """Can't update subscription on Adobe side."""
 
 
 class AdobeHttpError(AdobeError):
-    def __init__(self, status_code: int, content: str):
+    """Basic Adobe API HTTP error."""
+
+    def __init__(self, status_code: int, response_content: str):
+        """
+        Init Adobe HTTP Error.
+
+        Args:
+            status_code: Http error code.
+            response_content: Http content.
+        """
         self.status_code = status_code
-        self.content = content
-        super().__init__(f"{self.status_code} - {self.content}")
+        self.response_content = response_content
+        super().__init__(f"{self.status_code} - {self.response_content}")
 
 
 class AdobeAPIError(AdobeHttpError):
+    """Adobe API error."""
+
     def __init__(self, status_code: int, payload: dict) -> None:
+        """
+        Init AdobeAPIError.
+
+        Args:
+            status_code: Http error code.
+            payload: payload.
+        """
         super().__init__(status_code, json.dumps(payload))
         self.payload: dict = payload
         # 504 error response doesn't follow the expected format -
         # it uses "error_code" field instead of "code"
-        self.code: str = payload.get("code") or payload.get("error_code") or payload.get("error")
+        self.code: str | None = payload.get("code")
+        if self.code is None:
+            self.code = payload.get("error_code")
+        if self.code is None:
+            self.code = payload.get("error")
+
```

Comment:
Just confirming if the logic is ok, the condition in the old code consider "", None and False as falsy but the new one only checks for None.